### ISSUE TITLE
Dupe with crafting while in an inventory other than the player's

### ISSUE THREAD

**Issue by DWX12** [Tschipp/PrimitiveCrafting#10]
Using the most recent version of PrimitiveCrafting: PrimitiveCrafting+MC1.12.2+v1.3.1.

1. Open an inventory such as a chest
2. Take two items (such as two wood planks) from a stack
3. Return unused items to inventory
4. Craft items in player inventory area while still in chest
5. Close out of inventory. Items should be ejected onto the ground and crafted items will persist in inventory.

**Comment by DWX12** [comment]
Okay, it seems that just crafting items doesn't use up the ingredients period. I don't know what's going on or if I messed up some config somewhere.

I have OP status, but am in survival mode. I will see if anyone else on our small private server encounters this.

**Comment by nyanlathotep** [comment]
I'm getting the same issue, crafting doesn't use ingredients at all. I've tested on forge 2808 and 2768. The latter environment I stripped down to just PrimitiveCrafting+MC1.12.2+v1.3.1 running in its default config. When I attempt to craft something, I get an exception on the server, but it works in singleplayer, even in creative.

```
[00:09:39] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_121]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_121]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at tschipp.primitivecrafting.network.Craft$1.run(Craft.java:46) ~[Craft$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_121]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 5 more
```

**Comment by Tschipp** [comment]
Either I can't recreate this, or I'm understanding it wrongly:
[gif](https://i.gyazo.com/7056d3b4018b3be4855e50f621cd258a.gif)

**Comment by nyanlathotep** [comment]
Set up a new testbed to be sure, on a client and server with only forge 2768 and PrimitiveCrafting+MC1.12.2+v1.3.1, I get [this behavior](https://giant.gfycat.com/WebbedIncredibleAbalone.webm) and the server throws the exception I added above whenever I perform a primitive craft. The client alone in singleplayer is fine.

**Comment by Tschipp** [comment]
Oh okay I see, I will try that out!

**Comment by Tschipp** [comment]
Ok, should be fixed in 1.3.4
